src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -607,15 +606,32 @@ public Snapshot makeSnapshotFromSnapshotRequest(SnapshotRequestModel snapshotReq
         String snapshotQuery = queryModel.getQuery();
         Query query = Query.parse(snapshotQuery);
         String datasetName = query.getDatasetName();
-        createSnapshotTablesFromDatasetAsset(datasetName, assetName, snapshotSource, snapshot);
+        Dataset queryDataset = datasetService.retrieveByName(datasetName);
+        AssetSpecification queryAssetSpecification =
+            getAssetByNameFromDataset(queryDataset, assetName);
+        snapshotSource.assetSpecification(queryAssetSpecification);
+        conjureSnapshotTablesFromAsset(snapshot, snapshotSource);
       }
       case BYROWID -> {
         SnapshotRequestRowIdModel requestRowIdModel = requestContents.getRowIdSpec();
         conjureSnapshotTablesFromRowIds(requestRowIdModel, snapshot, snapshotSource);
       }
       case BYREQUESTID -> {
         String datasetName = snapshotRequestModel.getContents().get(0).getDatasetName();
-        createSnapshotTablesFromDatasetAsset(datasetName, ASSET_NAME, snapshotSource, snapshot);
+        Dataset queryDataset = datasetService.retrieveByName(datasetName);
+        UUID accessRequestId =
+            snapshotRequestModel.getContents().get(0).getRequestIdSpec().getSnapshotRequestId();
+        SnapshotAccessRequestResponse accessRequest = getById(accessRequestId);
+
+        AssetSpecification queryAssetSpecification =
+            buildAssetFromSnapshotAccessRequest(queryDataset, accessRequest);
+        // Make sure the AssetSpecification ID field is null
+        // so that it is not written to the database in the snapshot source table
```

Comment:
Where is the null check that enforces this? Does `buildAssetFromSnapshotAccessRequest()` return an specification with a non-null ID, and if so, where does it get it from?

Reply:
buildAssetFromSnapshotAccessRequest() returns a null id. So, this should never be populated. My thought was that this was more for documentation proposes rather than a real check. I ended up removing it, but glad to revisit it.